scalding-serialization/src/main/scala/com/twitter/scalding/serialization/OrderedSerialization.scala

```diff
@@ -101,6 +101,30 @@ object OrderedSerialization {
     case NonFatal(e) => CompareFailure(e)
   }
 
+  def viaTransform[T, U](
+    packFn: T => U,
+    unpackFn: U => T)(implicit otherOrdSer: OrderedSerialization[U]): OrderedSerialization[T] =
+    new OrderedSerialization[T] {
+
+      override def hash(t: T) = otherOrdSer.hash(packFn(t))
```

Comment:
I wonder if we should implement a simple cache since we might call write, hash, and dynamicSize in quick succession.

Something like:
```scala
private[this] var cache: (T, U) = null
private def packCache(t: T): U = {
  val readCache = cache
  if (null == readCache || !otherOrdSer.equiv(readCache._1, t)) {
    val u = packFn(t)
    cache = (t, u)
    u
  }
  else {
    readCache._2
  }
}
```